controllers/ingress.go

```diff
@@ -51,26 +53,41 @@ func (r *GslbReconciler) ensureIngress(instance *k8gbv1beta1.Gslb, i *v1beta1.In
 		if err != nil {
 			// Creation failed
 			log.Error(err, "Failed to create new Ingress", "Ingress.Namespace", i.Namespace, "Ingress.Name", i.Name)
-			return &reconcile.Result{}, err
+			return err
 		}
 		// Creation was successful
-		return nil, nil
+		return nil
 	} else if err != nil {
 		// Error that isn't due to the service not existing
 		log.Error(err, "Failed to get Ingress")
-		return &reconcile.Result{}, err
+		return err
 	}
 
 	// Update existing object with new spec and annotations
-	found.Spec = i.Spec
-	found.Annotations = i.Annotations
-	err = r.Update(context.TODO(), found)
-
-	if err != nil {
-		// Update failed
-		log.Error(err, "Failed to update Ingress", "Ingress.Namespace", found.Namespace, "Ingress.Name", found.Name)
-		return &reconcile.Result{}, err
+	if !ingressCompare(found, i) {
+		found.Spec = i.Spec
+		found.Annotations = utils.SaveAnnotations(found.Annotations, i.Annotations)
+		err = r.Update(context.TODO(), found)
+		if errors.IsConflict(err) {
+			r.Log.Info("Ingress has been modified outside of controller, retrying reconciliation",
+				"Ingress.Namespace", found.Namespace, "Ingress.Name", found.Name)
+			return nil
+		}
+		if err != nil {
+			// Update failed
+			log.Error(err, "Failed to update Ingress", "Ingress.Namespace", found.Namespace, "Ingress.Name", found.Name)
+			return err
+		}
 	}
 
-	return nil, nil
+	return nil
+}
+
+func ingressCompare(found *v1beta1.Ingress, i *v1beta1.Ingress) bool {
```

Comment:
Consider renaming
```suggestion
func ingressEqual(ing1 *v1beta1.Ingress, ing2 *v1beta1.Ingress) bool {
```
as we're testing for equality, not comparing entities.

Reply:
Just thinking about new package `metav1` next to the `utils` package and reuse ApiMachinery
https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#SetMetaDataAnnotation but in k8gb


Having `metav1.AnnotationsEqual(obj1 *ObjectMeta, obj2 *ObjectMeta)`, `metav1.MergeAnnotations(obj1 *ObjectMeta, obj2 *ObjectMeta)`

Reply:
Then we can reuse for any kind. (Maybe move to AbsaOSS/gopkg)

Reply:
Like here https://github.com/AbsaOSS/k8gb/pull/314 ?

Reply:
@somaritane yes

Reply:
Might be good in general along with reusing helpers from https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1.
I don't have strong opinion here, perhaps we can extract it later when we find more use in other projects.
@ytsarev, @k0da, wdyt?

Reply:
Good idea, though I wouldn't block this specific PR with it. We should log the Issue for that though

Reply:
ok, will provide in later.